src/gas-price/gas-price.ts

```diff
@@ -161,15 +161,16 @@ export const getRecommendedGasPrice = async (
   const lastUpdateTimestamp = sponsorLastUpdateTimestamp[sponsorWalletAddress];
 
   // Check if the next update is a retry of a pending transaction and if it has been pending longer than scalingWindow
-  if (lastUpdateTimestamp && lastUpdateTimestamp < Math.floor(Date.now() / 1000) - scalingWindow) {
+  if (lastUpdateTimestamp) {
+    const pendingPeriod = Math.floor(Date.now() / 1000) - lastUpdateTimestamp;
     const multiplier = calculateScalingMultiplier(
       recommendedGasPriceMultiplier,
       maxScalingMultiplier,
-      Math.floor(Date.now() / 1000) - lastUpdateTimestamp,
+      pendingPeriod,
       scalingWindow
     );
 
-    logger.warn('Scaling gas price.', { gasPrice: gasPrice.toString(), multiplier });
+    logger.warn('Scaling gas price.', { gasPrice: gasPrice.toString(), multiplier, pendingPeriod });
```

Comment:
I've added a `pendingPeriod` field, because it's quite important to know for how long the transaction is pending.